compiler/rustc_expand/src/base.rs

```diff
@@ -711,6 +711,9 @@ pub struct SyntaxExtension {
     /// Built-in macros have a couple of special properties like availability
     /// in `#[no_implicit_prelude]` modules, so we have to keep this flag.
     pub builtin_name: Option<Symbol>,
+    /// Rule spans. Used for linting. If the macro is not made up of rules,
+    /// it's empty.
+    pub rule_spans: Vec<Span>,
```

Comment:
So my main concern was that this data doesn't belong to `SyntaxExtension` in general, only to declarative macros.

Could you return `rule_spans` separately from `compile_(declarative_)macro` instead of merging it to `SyntaxExtension`?